pkg/operation/botanist/component/kubeproxy/kube_proxy_test.go

```diff
@@ -334,6 +338,216 @@ metadata:
   name: ` + configMapCleanupScriptName + `
   namespace: kube-system
 `
+
+			daemonSetNameFor = func(pool WorkerPool) string {
+				return "kube-proxy-" + pool.Name + "-v" + pool.KubernetesVersion
+			}
+			daemonSetYAMLFor = func(pool WorkerPool, ipvsEnabled, vpaEnabled bool) string {
+				out := `apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  annotations:`
+
+				if ipvsEnabled {
+					out += `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapNameFor(ipvsEnabled)) + `: ` + configMapNameFor(ipvsEnabled) + `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapConntrackFixScriptName) + `: ` + configMapConntrackFixScriptName + `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapCleanupScriptName) + `: ` + configMapCleanupScriptName + `
+    ` + references.AnnotationKey(references.KindSecret, secretName) + `: ` + secretName
+				} else {
+					out += `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapConntrackFixScriptName) + `: ` + configMapConntrackFixScriptName + `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapCleanupScriptName) + `: ` + configMapCleanupScriptName + `
+    ` + references.AnnotationKey(references.KindConfigMap, configMapNameFor(ipvsEnabled)) + `: ` + configMapNameFor(ipvsEnabled) + `
+    ` + references.AnnotationKey(references.KindSecret, secretName) + `: ` + secretName
+				}
+
+				out += `
+  creationTimestamp: null
+  labels:
+    gardener.cloud/role: system-component
+    origin: gardener
+  name: ` + daemonSetNameFor(pool) + `
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: kubernetes
+      pool: ` + pool.Name + `
+      role: proxy
+      version: ` + pool.KubernetesVersion + `
+  template:
+    metadata:
+      annotations:`
+
+				if ipvsEnabled {
+					out += `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapNameFor(ipvsEnabled)) + `: ` + configMapNameFor(ipvsEnabled) + `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapConntrackFixScriptName) + `: ` + configMapConntrackFixScriptName + `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapCleanupScriptName) + `: ` + configMapCleanupScriptName + `
+        ` + references.AnnotationKey(references.KindSecret, secretName) + `: ` + secretName
+				} else {
+					out += `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapConntrackFixScriptName) + `: ` + configMapConntrackFixScriptName + `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapCleanupScriptName) + `: ` + configMapCleanupScriptName + `
+        ` + references.AnnotationKey(references.KindConfigMap, configMapNameFor(ipvsEnabled)) + `: ` + configMapNameFor(ipvsEnabled) + `
+        ` + references.AnnotationKey(references.KindSecret, secretName) + `: ` + secretName
+				}
```

Comment:
Also, in the if-else, only the order of annotations is different or?

Reply:
> can't we make these repeated lines a function and use utils.Indent?

done

> Also, in the if-else, only the order of annotations is different or?

Yes, since the checksums are different in this case.